symbolic-debuginfo/src/macho.rs

```diff
@@ -41,6 +42,7 @@ impl MachError {
 pub struct MachObject<'d> {
     macho: mach::MachO<'d>,
     data: &'d [u8],
+    bcsymbolmap: Option<BCSymbolMap>,
```

Comment:
As discussed offline, this should be supplied to the debug session, instead. Contrary to `ObjectLike`, which provides a more-or-less cheap view on the file, `DebugSession` is meant for keeping state and performing lookups and resolution.

This also means that we need to add a symbol iterator to the debug session, which in all cases other than MachO is the same as `*Object::symbols`.

Reply:
from offline discussion: we have some ideas about what this api would look like but would like to have it validated with other auxiliary files.  the api introduced here is forwards compatible with such an api on the DebugSession so we'll go with this minimal version for now while we build up experience.